flake-module.nix

```
@@ -113,6 +131,13 @@ in
                 default = { };
                 type = hlintCheckSubmodule;
               };
+              packages = mkOption {
+                type = types.attrsOf packageAttrsSubmodule;
```

```suggestion
                type = types.lazyAttrsOf packageAttrsSubmodule;
```

Typical infinite recursion footgun, if packages depend on packages. I hope this was your problem?

---

Unfortunately not; seems there's another bug elsewhere. I'll push my test-case and then we can take a look.